### DISCUSSION TITLE
useDocumentDrawer from within a custom component

### DISCUSSION THREAD

**Discussion by gamellis** [payloadcms/payload#2527] [General]
Hey! I've created a custom component that shows a list of items from another collection. I want to use the document drawer to allow the user to read/edit the items in the list. How would I go about including the drawer within the custom component? I could include a link to leave the document and open the item, but using the drawer and keeping the same context would be sweet.

**Comment by aaronksaunders** [comment]
did you ever get this working? I have the drawers working but i need to pass data into the drawers and haven't figure that out yet.

**Reply by gamellis** [reply]
I didn't. I would be interested to see a POC though!

**Reply by aaronksaunders** [reply]
Once I get it working, the passing data to the drawer, I will post the code

**Reply by yfi** [reply]
Not sure if this has been solved, but I had this problem as well. I ended up wrapping the documentDrawer inside a context provider. Inside the drawer, custom field components grab the data with useContext and populate themselves.  This is cool, now I can create pre-populated child documents from within a parent document.

**Reply by MarkAtOmniux** [reply]
Hey folks - this hook is used internally by Payload, so you can copy the implementation they have and modify it for your use case :)

For example: https://github.com/payloadcms/payload/blob/13f71ac475a6e5f26e8e366447560d8373ecd71c/packages/payload/src/admin/components/forms/field-types/Upload/Input.tsx

**Comment by aaronksaunders** [comment]
Do u have any code u can share? How did you populate the context? Did u
have to create a whole custom page? At what level did u wrap the context?


--

Aaron K. Saunders
CEO Clearly Innovative Inc
***@***.***
www.clearlyinnovative.com


This email message and any attachment(s) are for the sole use of the
intended recipient(s) and may contain proprietary and/or confidential
information which may be privileged or otherwise protected from disclosure.
Any unauthorized review, use, disclosure or distribution is prohibited. If
you are not the intended recipient(s), please contact the sender by reply
email and destroy the original message and any copies of the message as
well as any attachment(s) to the original message.
